api/src/utils/generic/base.service.ts

```
@@ -0,0 +1,329 @@
+  }
+
+  /**
+   * Soft delete multiple entities on cascade
+   */
+  async deleteManyOnCascade(
+    where: FindOptionsWhere<T> | FindOptionsWhere<T>[],
+    relations: string[] = [],
+    userId?: string,
+  ): Promise<T[]> {
+    const relationsObject = relations.reduce(
+      (acc, rel) => ({ ...acc, [rel]: true }),
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)